ci/snap/ci-deploy.sh

```diff
@@ -16,23 +16,13 @@ sudo lxd init --minimal
 # `--use-lxd` applied to use a LXD container instead of a VM, to work around lack of support for KVM on CircleCI VMs.
 snapcraft --use-lxd
 
-if [[ ${1-} == "--dry-run" ]]; then
+if [[ ${1-} == "--dry-run" || ${CIRCLE_BRANCH-} != "master" ]]; then
   ! command -v git-machete
   sudo snap install git-machete*.snap --dangerous --classic
   git machete version
   sudo snap remove git-machete
 else
-#  echo "$SNAPCRAFT_LOGIN_CREDENTIALS_CONTENTS_BASE64" | base64 -d > ~/.snapcraft.login
-#  cat /home/circleci/.cache/snapcraft/log/snapcraft-20220708-133250.692570.log
   export SNAPCRAFT_STORE_CREDENTIALS=$SNAPCRAFT_LOGIN_CREDENTIALS_CONTENTS_BASE64
```

Comment:
Huh wait... so the creds in `SNAPCRAFT_STORE_CREDENTIALS` are expected in base64 by `snapcraft upload`? 😯

Reply:
@PawelLipski The overall problem wasn't with the encoding but with the way you login, previous approach needed explicit logging in with `snapcraft login --with <credentials-filename>` but in the new version it's not required, or to be even more precise, impossible to login explicitly which i tried to do since the documentation did not mention it. I guess the way it works right now is that you get logged in when executing a command that requires it, e.g. `snapcraft upload`.

Reply:
Okay... but still, AFAICS now the creds passed via `SNAPCRAFT_STORE_CREDENTIALS` env var are provided in base64 (unlike what was with `~/.snapcraft.login`), amirite?

Reply:
per documentation, yes: `The format of exported credentials differ between Snapcraft versions 6 and 7+. Snapcraft 6 exports a decrypted macaroon, which should not be shared, while later releases output a raw block of base64 encoded text that can only be used with Snapcraft 7.0 or greater.`